File: Assignment1/packet.py
class Packet:
  '''Generates and Breaks down a Packet'''

  def generate(self, content, body):
    content = content.split(" ")
    header = content[0]
    data = ""
    for l in content[1:]:
      data = data + l + "&" 
    return header + "\r\n" + data + "\n\n" + body

  def divide(self, data):
    header, content = data.split("\r\n", 1)
    content, body = content.split("\n\n", 1)
    return header, content, body

File: Assignment3/Assignment3/app/src/main/java/edu/cse/nolanburfield/assignment3/Packet.java
package edu.cse.nolanburfield.assignment3;

/**
 * Created by nolanburfield on 4/12/15.
 */

import android.util.Log;

public class Packet {

    private String header;
    private String data;
    private String body;

    private static final String TAG = "packet";

    public Packet(String header, String data, String body) {
        this.header = header;
        this.data = data;
        this.body = body;
        Log.v(TAG, "Packet Created: " + this.header + " " + this.data + " " + this.body);
    }

    public String send() {
        Log.v(TAG, "Packet " + this.header + " Returned.");
        return this.header + "\r\n\r" + this.data + "\r\n\r" + this.body;
    }

    public void result(String value) {
        String[] data;
        data = new String[3];
        data = value.split("\r\n\r");
        this.header = data[0];
        this.data = data[1];
        this.body = data[2];
    }

    public String getHeader() {
        return this.header;
    }

    public String getData() {
        return this.data;
    }

    public String getBody() {
        return this.body;
    }
}
